caas/scripts.go

```diff
@@ -27,12 +27,21 @@ cp %[3]s/jujud $JUJU_TOOLS_DIR/jujud
 `[1:]
 
 	// MongoStartupShTemplate is used to generate the start script for mongodb.
+	// Mongo is very specific about what permissions the shared secret must have,
+	// so we must copy it and lock it down for rootless k8s controllers.
 	MongoStartupShTemplate = `
 args="%s"
 ipv6Disabled=$(sysctl net.ipv6.conf.all.disable_ipv6 -n)
 if [ $ipv6Disabled -eq 0 ]; then
   args="${args} --ipv6"
 fi
+SHARED_SECRET_SRC="%s"
+SHARED_SECRET_DST="%s"
+rm "${SHARED_SECRET_DST}" || true
+cp "${SHARED_SECRET_SRC}" "${SHARED_SECRET_DST}"
+chown 170:170 "${SHARED_SECRET_DST}"
```

Comment:
Can we make this dynamic by injecting the data or putting a comment saying this needs to be updated in several places?